hydrolib/profile_optimizer/profile_optimizer/preprocessing.py

```diff
@@ -0,0 +1,173 @@
+#Inladen packages
+import pandas as pd
+import geopandas as gpd
+import xarray as xr
+from shapely.geometry import LineString
+from shapely.errors import ShapelyDeprecationWarning
+from pathlib import Path
+import numpy as np
+import warnings
+from hydrolib.core.io.crosssection.models import CrossLocModel
+from sympy import solve
+from sympy import Symbol
+import math
+
+
+#Functie voor het inladen van het te optimaliseren gebied in de vorm van een shapefile
+def select_crosssections(shapefile_path, model_network_nc, crossdef_filename, output_folder):
+    output_folder = Path(output_folder)
+    shapefile_area = gpd.read_file(shapefile_path)
+    ds = xr.open_dataset(model_network_nc)
+
+    #maak een geodataframe van alle nodes
+    df = pd.concat([pd.Series(ds['network1d_geom_x'].values), pd.Series(ds['network1d_geom_y'].values)], axis=1)
+    df.columns = ['network1d_geom_x', 'network1d_geom_y']
+    gdf = gpd.GeoDataFrame(df, geometry=gpd.points_from_xy(df.network1d_geom_x, df.network1d_geom_y))
+    gdf.to_file(driver = 'ESRI Shapefile',filename=str(output_folder/'network1d_geomxandy.shp'))
+
+    #aantal nodes per branch id
+    df_branches = pd.concat([pd.Series(ds['network1d_geom_node_count'].values), pd.Series(ds['network1d_branch_id'].values.astype(str))], axis=1)
+    df_branches.columns = ['network1d_geom_node_count', 'branchid']
+
+    #maak aparte linestring aan voor elke branch id met alle nodes die erbij horen
+    df_branches['line_geometry'] = ''
+    df_branches['start_node'] = ''
+    df_branches['end_node'] = ''
+
+    for j in range(len(df_branches)):
+        if j == 0:
```

Comment:
Here again you can reduce code duplicity by:
```
   if j == 0:
            start_node = 0
            end_node = 0 + df_branches['network1d_geom_node_count'][j]
    else:
            start_node = df_branches['network1d_geom_node_count'][:j].sum()
            end_node = start_node + df_branches['network1d_geom_node_count'].iloc[j]
    ... and then the rest
```

Reply:
This was duplicate code anyway, whoops! Deleted this whole function